src/account/PasskeyAccount.sol

```diff
@@ -0,0 +1,260 @@
+    }
+
+    function validateSignature(
+        UserOperation calldata userOp,
+        bytes32 userOpHash // As Webauthn's challenge
+    ) external view returns (uint256) {
+        return _validateSignature(userOp, userOpHash);
+    }
+
+    function _initPasskey(
+        string memory credId,
+        uint256 pubKeyX,
+        uint256 pubKeyY
+    ) internal {
+        require(
+            passkey.pubKeyX == 0 && passkey.pubKeyY == 0,
+            "Passkey already exists"
+        );
+        passkey.credIdHash = keccak256(abi.encodePacked(credId));
+        passkey.pubKeyX = pubKeyX;
+        passkey.pubKeyY = pubKeyY;
+        emit PasskeyInitialized(credId, passkey.pubKeyX, passkey.pubKeyY);
+    }
+
+    /**
+     * @param userOp typical userOperation
+     * @param userOpHash the hash of the user operation.
+     * @return validationData
+     */
+    function _validateSignature(
+        UserOperation calldata userOp,
+        bytes32 userOpHash // As Webauthn's challenge
+    ) internal view virtual override returns (uint256) {
+        require(
+            passkey.pubKeyX != 0 && passkey.pubKeyY != 0,
+            "Passkey doesn't exist"
+        );
+        (
+            bytes memory authenticatorData,
+            bool requireUserVerification,
+            string memory clientDataJSON,
+            uint256 challengeLocation,
+            uint256 responseTypeLocation,
+            uint256 r,
+            uint256 s
+        ) = abi.decode(
+            userOp.signature,
+            (bytes, bool, string, uint256, uint256, uint256, uint256)
+        );
+        return _verifySignatureWebauthn(
+            abi.encodePacked(userOpHash),
+            authenticatorData,
+            requireUserVerification,
+            clientDataJSON,
+            challengeLocation,
+            responseTypeLocation,
+            r,
+            s,
+            passkey.pubKeyX,
+            passkey.pubKeyY
+        ) ? 0 : SIG_VALIDATION_FAILED;
+    }
+
+    function _verifySignatureAllowMalleability(
+        bytes32 messageHash,
+        uint256 r,
+        uint256 s,
+        uint256 x,
+        uint256 y
+    ) internal view returns (bool) {
+        bytes memory args = abi.encode(messageHash, r, s, x, y);
+        (bool success, bytes memory ret) =
+            address(p256Verifier).staticcall(args);
+        assert(success); // never reverts, always returns 0 or 1
+
+        return abi.decode(ret, (uint256)) == 1;
+    }
+
+    /// P256 curve order n/2 for malleability check
+    uint256 constant P256_N_DIV_2 =
+        57896044605178124381348723474703786764998477612067880171211129530534256022184;
+
+    function _verifySignature(
+        bytes32 messageHash,
+        uint256 r,
+        uint256 s,
+        uint256 x,
+        uint256 y
+    ) internal view returns (bool) {
+        // check for signature malleability
+        if (s > P256_N_DIV_2) {
+            return false;
+        }
```

Comment:
Q: Shall we alter the `s` when it is greater than `P256_N_DIV_2`, or signature from webauthn might fail irregularly?

Reply:
Or we would enforce client to do this?

Reply:
It should be mandated for the client to follow this approach. If the condition 'if(s>n/2) n-s' is hard-coded in the contract, and a malicious value of s is provided by the client, this hard-coded logic will transform s into a verifiably passable form.